### PR TITLE
Fixup Party min/max

### PR DESCRIPTION
**Pull request by AustinLMayes**
Ensure party max is always at least or greater than the minimum needed players.

### CODE REVIEW

File: core/src/main/java/tc/oc/pgm/teams/TeamModule.java [hunk 1]
```diff
@@ -135,6 +135,10 @@ private static TeamFactory parseTeamDefinition(Element el, MapFactory factory)
       throw new InvalidXMLException("Max overfill can not be less then max players.", el);
     }
 
+    if (maxPlayers <= minPlayers) {
+      throw new InvalidXMLException("Min players can not be less (or equal) to max players.", el);
```
Comment (Pablete1234) [inline core/src/main/java/tc/oc/pgm/teams/TeamModule.java:139]: Min can definetly be equal to max.
If you want a map to play 8v8, with no more and no less players, you set both min and max to 8 in each team

File: core/src/main/java/tc/oc/pgm/ffa/FreeForAllModule.java [hunk 2]
```diff
@@ -86,6 +86,15 @@ public FreeForAllModule parse(MapFactory factory, Logger logger, Document doc)
           colors = XMLUtils.parseBoolean(Node.fromAttr(elPlayers, "colors"), colors);
         }
 
+        if (minPlayers > maxPlayers) {
+          if (elPlayers != null)
+            throw new InvalidXMLException("\"max\" must be greater than \"min\"", elPlayers);
```
Comment (Pablete1234) [inline core/src/main/java/tc/oc/pgm/ffa/FreeForAllModule.java:91]: max can also be equal to min, not necessarily greater

File: core/src/main/java/tc/oc/pgm/ffa/FreeForAllModule.java [hunk 3]
```diff
@@ -86,6 +86,15 @@ public FreeForAllModule parse(MapFactory factory, Logger logger, Document doc)
...
+          else {
+            if (maxPlayers > 1) minPlayers = maxPlayers / 2;
```
Comment (Pablete1234) [inline core/src/main/java/tc/oc/pgm/ffa/FreeForAllModule.java:93]: I belive it makes more sense to force minPlayers to maxPlayers (bukkit max) if it goes over the limit. That's the biggest minimum that will be able to be met

**Review by Pablete1234 — CHANGES_REQUESTED** [review]

File: core/src/main/java/tc/oc/pgm/ffa/FreeForAllModule.java (hunk 3 above)
Comment (Electroid) [reply core/src/main/java/tc/oc/pgm/ffa/FreeForAllModule.java:93]: Same, PGM should not care about `Bukkit#getMaxPlayers` anywhere.

**Comment by Electroid** [comment]
Closing, this should be done with XML parsing ranges.

**Pull request closed without merging**